ArduCopter/mode.h

```diff
@@ -1254,3 +1254,51 @@ class ModeFollow : public ModeGuided {
 
     uint32_t last_log_ms;   // system time of last time desired velocity was logging
 };
+
+class ModeZigzag : public Mode {        //a new mode for zigzag
+
+public:
+
+    // inherit constructor
+    using Copter::Mode::Mode;
+
+    bool init(bool ignore_checks) override;
+    void run() override;
+
+    bool requires_GPS() const override { return true; }
+    bool has_manual_throttle() const override { return false; }
+    bool allows_arming(bool from_gcs) const override { return false; }
+    bool is_autopilot() const override { return true; }
+
+    void zigzag_receive_signal_from_auxsw(uint8_t aux_switch_position);
```

Comment:
We have a specific enumeration for the switch position ->do you mean I should use RC_Channel::aux_switch_pos_t instead of uint_t? Sorry if the question is stupid, my C++ needs improving...

Reply:
@caomuqing Correct!

Reply:
When I try to use RC_Channel::aux_switch_pos_t I get error that 
> 'typedef enum RC_Channel::aux_switch_pos RC_Channel::aux_switch_pos_t' is protected

Reply:
Ah - that's my bad, sorry.

Move that enumeation to be public, please.  I'll make the same change in master at some stage.

Reply:
OK, I have moved the enumeration in RC_Channel library to public